src/plugins/core/merge.ts

```diff
@@ -155,6 +159,29 @@ export class MergePlugin extends CorePlugin<MergeState> implements MergeState {
       .filter(isDefined);
   }
 
+  /**
+   * Same as `getRangeString` but add all necessary merge to the range to make it a valid selection
+   */
+  getSelectionRangeString(range: Range, forSheetId: UID): string {
```

Comment:
> As far as I can tell, the range "plugin" RangeAdapter (which is actually not a plugin) only uses its getters in its own getters.

this is not true unfortunately, we rely on the getters to adapt the ranges (though it's just in one case atm, {**hard**} maybe we could get rid of it. that being said, I don't see how it would solve the dependency cycle issue :/

Reply:
you're right :disappointed: